service/history/api/startworkflow/api.go

```diff
@@ -348,6 +347,10 @@ func (s *Starter) createAsCurrent(
 	creationParams *creationParams,
 	currentWorkflowConditionFailed *persistence.CurrentWorkflowConditionFailedError,
 ) error {
+	// TODO(stephanos): remove this hack - this is here for Update-with-start to reapply the Update
+	if _, err := s.createLeaseFn(creationParams.workflowLease, s.shardContext, nil); err != nil {
+		return err
+	}
```

Comment:
Note that his will now call `NewWorkflowLeaseAndContext` twice for a regular workflow start. But that should be fine, as it has no side effects.